net/art/stride_table.go

```diff
@@ -69,61 +69,50 @@ const (
 	lastHostIndex = 0b1_1111_1111
 )
 
-// getChild returns the child strideTable pointer for addr (if any), and an
-// internal array index that can be used with deleteChild.
-func (t *strideTable[T]) getChild(addr uint8) (child *strideTable[T], idx int) {
-	idx = hostIndex(addr)
-	return t.entries[idx].child, idx
+// getChild returns the child strideTable pointer for addr (if any).
+func (t *strideTable[T]) getChild(addr uint8) *strideTable[T] {
+	return t.children[addr]
 }
 
-// deleteChild deletes the child strideTable at idx (if any). idx should be
-// obtained via a call to getChild.
-func (t *strideTable[T]) deleteChild(idx int) {
-	t.entries[idx].child = nil
+// deleteChild deletes the child strideTable at addr (if any).
+func (t *strideTable[T]) deleteChild(addr uint8) {
+	t.children[addr] = nil
 	t.childRefs--
 }
 
 // setChild replaces the child strideTable for addr (if any) with child.
 func (t *strideTable[T]) setChild(addr uint8, child *strideTable[T]) {
-	t.setChildByIndex(hostIndex(addr), child)
-}
-
-// setChildByIndex replaces the child strideTable at idx (if any) with
-// child. idx should be obtained via a call to getChild.
-func (t *strideTable[T]) setChildByIndex(idx int, child *strideTable[T]) {
-	if t.entries[idx].child == nil {
+	if t.children[addr] == nil {
 		t.childRefs++
 	}
-	t.entries[idx].child = child
+	t.children[addr] = child
 }
 
 // getOrCreateChild returns the child strideTable for addr, creating it if
 // necessary.
 func (t *strideTable[T]) getOrCreateChild(addr uint8) (child *strideTable[T], created bool) {
-	idx := hostIndex(addr)
-	if t.entries[idx].child == nil {
-		t.entries[idx].child = &strideTable[T]{
+	if t.children[addr] == nil {
+		t.children[addr] = &strideTable[T]{
 			prefix: childPrefixOf(t.prefix, addr),
 		}
 		t.childRefs++
-		return t.entries[idx].child, true
+		return t.children[addr], true
```

Comment:
could use a local variable to avoid the map operation on this return, since you know the value from line 95

not sure if the compiler gets this right these days. at least in the past it did not.

Reply:
I've forgotten how to get the compiler to show its workings, so I made the lookup use a local.